src/components/uast/Node.js

```diff
@@ -0,0 +1,236 @@
+import React, { Component } from 'react';
+import styled from 'styled-components';
+import {eventHandler} from './EventHandler';
+import {font, border} from '../../styling/variables';
+
+const INDENT_SIZE = 20;
+const WHITE_SPACE = 5;
+const LINE_HEIGHT  = 29;
+
+const PROPERTY_VALUE_SEPARATOR = '\':\'';
+const STRING_LIMITER = '\'\\27\'';
+const COLLAPSIBLE_COLLAPSED = '\'+\'';
+const COLLAPSIBLE_EXTENDED = '\'-\'';
+
+const StyledItem = styled.div`
+  margin-left: ${INDENT_SIZE}px;
+  min-width: 400px;
+`;
+
+const StyledTitle = styled.div`
+  min-height: ${LINE_HEIGHT}px;
+
+  &::before {
+    content: '';
+    width: 15px;
+    display: inline-block;
+  }
+`;
+
+const StyledCollapsibleTitle = StyledTitle.extend`
+  cursor: pointer;
+
+  &::before {
+    content: ${props => props.collapsed ? COLLAPSIBLE_COLLAPSED : COLLAPSIBLE_EXTENDED};
+    color: ${props => props.collapsed ? font.color.accentDark : font.color.accentLight};
+  }
+
+  &:hover > span {
+    border-bottom: 1px dashed ${border.light};
+  }
+`;
+
+const StyledCollapsibleContent = styled.div`
+  display: ${props => props.collapsed ? 'none' : 'block'};
+  border-left: 1px solid ${border.smooth};
+  margin-left: 4px;
+`;
+
+const StyledLabel = styled.summary`
+  display: inline;
+  font-family: ${font.family.prose};
+  color: ${font.color.light};
+`;
+
+const StyledValue = styled.span`
+  color: ${
+    props => props.type === 'number' || props.type === 'object' ?
+      font.color.accentDark :
+      props.type === 'string' ? font.color.accentLight : font.color.dark
+    };
+
+  &::before{
+    content: ${props => props.type === 'string' ? STRING_LIMITER : ''};
+  }
+
+  &::after {
+    content: ${props => props.type === 'string' ? STRING_LIMITER : ''};
+  }
+`;
+
+const StyledPropertyName = styled.span`
+  margin-right: ${WHITE_SPACE}px;
+
+  &::after {
+    content: ${PROPERTY_VALUE_SEPARATOR};
+    color: ${font.color.light};
+  }
+`;
+
+function Value({value}) {
+  return (
+    <StyledValue type={typeof value}>{ value !== null ? value : 'null'}</StyledValue>
+  );
+}
+
+function Label({name}) {
+  return (
+    <StyledLabel>{name}</StyledLabel>
+  );
+}
+
+function PropertyName({name}) {
+  return (
+    name ?
+      <StyledPropertyName>{name}</StyledPropertyName> :
+      null
```

Comment:
If we are going to use multiline ternary operators, at least let's write them everytime the same way:

```js
cond
 ? if_true
 : if_false
```